wandb/util.py

```diff
@@ -1749,6 +1749,12 @@ def _is_offline() -> bool:
     ) or wandb.setup().settings._offline  # type: ignore
 
 
+def _is_offline_legacy() -> bool:
+    return (  # type: ignore[no-any-return]
+        wandb.run is not None and wandb.run.settings._offline_legacy
+    ) or wandb.setup().settings._offline_legacy  # type: ignore
```

Comment:
no no, the original utility is already an anti-pattern, let's not add more to it :(